apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransaction.cs

```diff
@@ -441,6 +441,8 @@ public new Task<DateTime> CommitAsync(CancellationToken cancellationToken = defa
             {
                 var callSettings = SpannerConnection.CreateCallSettings(settings => settings.CommitSettings, CommitTimeout, cancellationToken);
                 var response = await _session.CommitAsync(request, callSettings).ConfigureAwait(false);
+                // We dispose of the SpannerTransaction to inmediately release the session to the pool when possible.
```

Comment:
This feels vaguely odd - because the caller won't know whether it's been disposed or not. Presumably it's fine for the transaction to be disposed here *and* by the caller? I'm assuming so, in which case this is fine really, just feels odd.